lib/src/cmsd/CmsdHeaderField.ts

```diff
@@ -1,24 +1,30 @@
-// NOTE: String enum values cannot be "computed", so we need to redefine them here.
-// import { CMSD_DYNAMIC } from './CMSD_DYNAMIC.js';
-// import { CMSD_STATIC } from './CMSD_STATIC.js';
+import { CMSD_DYNAMIC } from './CMSD_DYNAMIC.js';
+import { CMSD_STATIC } from './CMSD_STATIC.js';
 
 /**
  * CMSD header fields.
  *
  * @group CMSD
  *
+ * @enum
+ *
  * @beta
  */
-export enum CmsdHeaderField {
+export const CmsdHeaderField = {
 	/**
 	 * Keys whose values persist over multiple requests for the object.
 	 */
-	STATIC = 'CMSD-Static',
+	STATIC: CMSD_STATIC as typeof CMSD_STATIC,
 
 	/**
 	 * Keys whose values apply only to the next transmission hop. Typically a
 	 * new CMSD-Dynamic header instance will be added by each intermediary
 	 * participating in the delivery.
 	 */
-	DYNAMIC = 'CMSD-Dynamic',
-}
+	DYNAMIC: CMSD_DYNAMIC as typeof CMSD_DYNAMIC,
+} as const;
+
+/**
+ * @internal
+ */
+export type CmsdHeaderField = typeof CmsdHeaderField[keyof typeof CmsdHeaderField];
```

Comment:
As @cjpillsbury suggested in his comment (https://github.com/streaming-video-technology-alliance/common-media-library/issues/103#issuecomment-2436051101), would it be possible to use `ValueOf` in place to avoid duplicated logic in every file?

Reply:
I wouldn't mind using a helper, but the `ValueOf` helper mentioned in that comment doesn't compile because of circular references, as well as a few others I've tried:
 
![Screenshot 2024-10-28 at 09 57 36](https://github.com/user-attachments/assets/79923df1-de9f-43fe-bd53-4a43be12bf6c)

If anyone has a solution for this I'd be open to adding it.

Reply:
@littlespex so it's an issue of type vs. value. You can do what you're trying, but I'd lean towards naming the type alias you're constructing differently. Defer to y'all on overall preference. Here's a simple writeup of two different options:

```ts
export const CmcdStreamingFormat = {
    DASH: 'd',
    HLS: 'h',
    SMOOTH: 's',
    OTHER: 'o',
};

// This is the type of the *complex object,* CmcdStreamingFormat
export type CmcdStreamingFormat = typeof CmcdStreamingFormat;

// This should probably be defined + exported in a shared/global types file for the project
export type ValueOf<T> = T[keyof T];

// This is the type of the *values* defined on the complex object, CmcdStreamingFormat
// You *could* do this, but it could get confusing because now your named type definition and your corresponding value declaration have "different shapes"
// Note the "typeof" here
// export type CmcdStreamingFormat = ValueOf<typeof CmcdStreamingFormat>;

// Or you could name the type declaration/alias for the values differently, e.g. making it plural, which allows you to have both types
// and have a corresponding type for the exported lookup/map object, CmcdStreamingFormat
export type CmcdStreamingFormats = ValueOf<CmcdStreamingFormat>;
```

Here's a TS Playground of ☝️:
https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBAYwgOwM7wMIFsEBMDKMUwAhlgJbIDmAYtFifALxwDeAUHF3ACICC+ABIAuOAHJcYgDSduggDL5RYgBbTZXfAFkA8joAqI8anXc4BwQFEASsojqAvgG527APRu4+leVRxfcDAqwIEAnmAhEABmgcFwAFRIWGAANqBwEABGAFbACDBS8XDYeITEZJS09IzsoJCwYRHFOAREpBTUdFAMzI3A0c2lbRWd1TAu7p7eAagqEACuKbhwYFBZJJkpoXCZIbjAUZTAywDUcHXQMMf+yHAkcLMkxLhuVCnrKX1+h2lwUdCxEKrCC5fK1cCXPpwABqJBS82AOiiAB59AA+OAsfQAbQA1sBQgN9ABdCYeLw+PwBIIhGDhSIxGkJABucIRqCK+0OyGuKEBiAgyTSIAyOTyBUGrXKHSq3Rq5IAmgsEkhFrhORBYr4pDt5vByPBVUs4FRgIaUFF5qhKjs8iQrSFkBAAO5wQnzKBwZBka50ppcygG8h8kjIZbuz1IKDEVCQMM21nwvZ5FJPRjB24qEjMkIAIlw5CiUWAxGQ8EeEVQucmcAAchArvzc37+lFc3BgsQaxcGi3JWV2pUuj1MTC2YiUS2BiUpYPRnKYGiyZ4dJ73QK1V6ffy+-sEKmoOmUG44eQSN8AUzE+y4AWiyXgGWtjrgAA6KivuAMXE2g0reGHikOrOj4CAqHcKTvM6fjrjAmpZjmOwNuBLaoDWobLAhIT3FGMZxgW1BQv8npMj2VzLO8EC4vMYBuAwYCiqCEozgOIyyj04L1PAfYscMMrDowfgsLCSZIsivHSkOYxLkAA

Reply:
Having different names for the type vs value is something we want to avoid.

Reply:
It would:
- Create a breaking semver change
- Force us to document the `'s'` types separetly
- Having to create in interim type just to use with the helper type seems to defeat the purpose of encapsulating the helper in the first place. 

As mentioned in the inciting issue, the "`keyof typeof`" approach is a bit verbose on the authoring side of things, but it makes things cleaner on the user side.

Reply:
Cool, was simply responding to 
> I wouldn't mind using a helper, but the ValueOf helper mentioned in that comment doesn't compile because of circular references, as well as a few others I've tried

Reply:
`ValueOf<typeof CmcdStreamingFormat>` is the answer. Updating now